projects/super_res/trainer_mod_flow.py

```diff
@@ -0,0 +1,66 @@
+import os
+
+from model.autoreg_diffusion_mod_flow import Unet, Flow, GaussianDiffusion, Trainer
+from data.load_data import load_data
+from config_mod_flow import config
+
+def main():
+    model = Unet(
+        dim = config.dim,
+        channels = 2 * config.data_config["img_channel"],
+        out_dim = config.data_config["img_channel"],
+        dim_mults = config.dim_mults,
+        learned_sinusoidal_cond = config.learned_sinusoidal_cond,
+        random_fourier_features = config.random_fourier_features,
+        learned_sinusoidal_dim = config.learned_sinusoidal_dim
+    ).cuda()
+
+    flow = Flow(
+        dim = config.dim,
+        channels = 3 * config.data_config["img_channel"],
+        out_dim = 2,
```

Comment:
So this is the 2D part?

Reply:
Flow is both in 2D and 3D, the out_dim is 2 in 2d, 3 in 3d. The way it is used to warp is using scale_space_warp in 3d and flow_warp in 2d